mcp_proxy_for_aws/sigv4_helper.py

```diff
@@ -195,7 +196,12 @@ async def _handle_error_response(response: httpx.Response) -> None:
             # Try to parse JSON error details
             error_details = response.json()
             logger.log(log_level, 'HTTP %d Error Details: %s', response.status_code, error_details)
-        except Exception:
+            if response.status_code == 401 or response.status_code == 403:
+                request = await response.request.aread()
+                request_body = json.loads(request.decode())
+                if isinstance(request_body, dict) and request_body.get('method') == 'initialize':
```

Comment:
Why is this thrown only on `initialize`? I presume because other MCP operations would normally return 200 if they are implemented according to the spec. But we can't make that assumption especially as the proxy can be leveraged by any non-AWS service specific MCP server (for example an AgentCore Runtime MCP server would be feasible to use with the proxy).